### PR TITLE
OSD-8117, OSD-8118: cluster login interactive & get alert metadata added

### PR DESCRIPTION
**Pull request by supreeth7**

### CODE REVIEW

File: cmd/pdcli/alerts/cmd.go [hunk 1]
```diff
@@ -32,71 +36,110 @@ import (
...
+	Cmd.Flags().BoolVar(&options.low, "low", false, "View all low alerts")
+	Cmd.Flags().BoolVar(&options.high, "high", true, "View all high alerts")
 
 	// Incident Assignment
 	Cmd.Flags().StringVar(
-		&args.assignment,
+		&options.assignment,
 		"assigned-to",
 		"self",
 		"Filter alerts based on user or team",
```
Comment (chopraapooja) [inline cmd/pdcli/alerts/cmd.go:81]: any example of user or team you can mention will be helpful

File: cmd/pdcli/alerts/cmd.go [hunk 2]
```diff
@@ -32,71 +36,110 @@ import (
...
 	)
 
 	// Columns displayed
 	Cmd.Flags().StringVar(
-		&args.columns,
+		&options.columns,
 		"columns",
-		"incident.id,name,cluster.id,status,severity",
+		"incident.id,cluster.name,alert,cluster.id,status,severity",
 		"Specify which columns to display separated by commas without any space in between",
```
Comment (chopraapooja) [inline cmd/pdcli/alerts/cmd.go:89]: specify the documentation link to view all possible columns

File: cmd/pdcli/alerts/cmd.go [hunk 3]
```diff
@@ -32,71 +36,110 @@ import (
...
 	)
+
+	// Interactive mode
+	Cmd.Flags().BoolVarP(
+		&options.interactive,
+		"interactive",
+		"i",
+		false,
+		"interactive alerts",
```
Comment (chopraapooja) [inline cmd/pdcli/alerts/cmd.go:98]: message could be something like, `Use interactive mode`

File: cmd/pdcli/alerts/cmd.go [hunk 4]
```diff
@@ -32,71 +36,110 @@ import (
...
-	// Get incident IDs
-	incidentIDs, err := GetIncidents(client)
+	// Check for incident ID argument
+	if len(args) > 0 {
+		incidentID = strings.TrimSpace(args[0])
+
+		// Validate the incident ID
+		match, _ := regexp.MatchString(constants.IncidentIdRegex, incidentID)
+
+		if !match {
```
Comment (chopraapooja) [inline cmd/pdcli/alerts/cmd.go:123]: inplace of `!match` you could use the err returned from regexp.MatchString

Comment (chopraapooja) [reply cmd/pdcli/alerts/cmd.go:123]: nitpick, but just as convension, err is more reliable and conventional

File: cmd/pdcli/alerts/cmd.go (hunk 2 above)
Comment (supreeth7) [reply cmd/pdcli/alerts/cmd.go:89]: It's in the README, wouldn't that suffice?

File: cmd/pdcli/alerts/cmd.go [hunk 5]
```diff
@@ -105,15 +148,25 @@ func alertsHandler(cmd *cobra.Command, args []string) error {
...
+		printAlerts(alerts)
+	}
 
 	return nil
 }
 
-// getIncidents returns a string slice consisting IDs of the first 10 incidents.
-func GetIncidents(c client.PagerDutyClient) ([]string, error) {
+// GetIncidents returns an array pagerduty incidents.
+func GetIncidents(c client.PagerDutyClient) ([]pdApi.Incident, error) {
```
Comment (chopraapooja) [inline cmd/pdcli/alerts/cmd.go:168]: I would keep this in a separate package(pdcli/incidents)  for reusability purpose, assuming the signature would be something like 
```func GetIncidents(c client.PagerDutyClient, options pdApi.ListIncidentsOptions) ([] pdApi.Incident, error)```

Comment (chopraapooja) [reply cmd/pdcli/alerts/cmd.go:168]: General feedback: Try to lighten up the weight for cmd.go, try to see which things you can decouple and organize

File: cmd/pdcli/alerts/cmd.go (hunk 4 above)
Comment (supreeth7) [reply cmd/pdcli/alerts/cmd.go:123]: `match` is validating the regex, so I would suggest using `match` as it is.
Tried using `err` and didn't get the expected output for the wrong incident ID.
```
./pdcli alerts P0X97E12
status code: 404
```

File: cmd/pdcli/alerts/cmd.go (hunk 5 above)
Comment (supreeth7) [reply cmd/pdcli/alerts/cmd.go:168]: Incidents and alerts are inter-chained, so wouldn't it be better to let `GetIncidents` be in the same file as alerts?

File: cmd/pdcli/alerts/cmd.go (hunk 1 above)
Comment (supreeth7) [reply cmd/pdcli/alerts/cmd.go:81]: The same is mentioned in the ReadMe too, wouldn't that do?
If `assigned-to=team`, it defaults to Platform-SRE for now, will change it in the teams ticket.

File: cmd/pdcli/alerts/cmd.go (hunk 5 above)
Comment (supreeth7) [reply cmd/pdcli/alerts/cmd.go:168]: Suppose I move it into a separate package, incidents are fetched based on the `pdcli alerts` options, replicating options struct in the other pkg, wouldn't that become more complex?

Comment (supreeth7) [reply cmd/pdcli/alerts/cmd.go:168]: Adding to that, to lighten up the weight, shall I move the select functions to a separate pdcli/prompt pkg? And the `clusterLogin` to pdlci/ocm?

File: cmd/pdcli/alerts/cmd.go (hunk 2 above)
Comment (chopraapooja) [reply cmd/pdcli/alerts/cmd.go:89]: For end users, to make it available in help message

**Pull request merged**